packages/opentelemetry-core/src/trace/TraceState.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+
+// TODO validate maximum number of items
+// const maximumTraceStateItems = 32;
```

Comment:
Why would this be limited? What if a vendor needs more than the limit?

Reply:
This limit is enforced by the specs, "_There can be a maximum of 32 list-members in a list._".

Reply:
I think it should be fine for now as 32 is somewhat high for this. It may be worth it however to start a discussion about the reasoning for the limit. Since there is already a maximum length for the whole state, it doesn't really make sense to add such an additional limit.

Reply:
@rochdev one vendor can only have one key-value pair.
Here is an explanation of how entries are added and replaced: https://github.com/w3c/trace-context/blob/master/spec/20-http_header_format.md#relationship-between-the-headers